assets/src/edit-story/components/canvas/eyedropperLayer.js

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useEffect, useRef } from 'react';
+import styled from 'styled-components';
+import { rgba, readableColor } from 'polished';
+
+/**
+ * Internal dependencies
+ */
+import { useCanvas, useLayout } from '../../app';
+import { FULLBLEED_RATIO } from '../../constants';
+import { Layer, PageArea } from './layout';
+
+const EyedropperBackground = styled(Layer)`
+  background: rgba(0, 0, 0, 0.7);
+  cursor: not-allowed;
+  z-index: 2;
+`;
+
+const DisplayPageArea = styled(PageArea)`
+  position: absolute;
+`;
+
+const EyedropperCanvas = styled.div`
+  background: transparent;
+  border-radius: 5px;
+  color: black;
+  position: absolute;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  right: 0;
+  cursor: default;
+`;
+
+const ColorInfo = styled.div`
+  position: absolute;
+  text-transform: uppercase;
+  transform: translateX(-50%);
+  padding: 9px;
+  min-width: 102px;
+  text-align: center;
+`;
+
+const ZoomCanvas = styled.canvas`
+  border: 2px solid black;
+  border-radius: 50%;
+  transform: translate(calc(-50% + 4px), 20px);
+  background: ${({ theme }) => theme.colors.bg.primary};
+`;
+
+const Zoom = styled.div`
+  position: absolute;
+  display: flex;
+  top: 0;
+  left: 0;
+  transform: translateY(-1000px);
+  pointer-events: none;
+`;
+
+function EyedropperLayer() {
+  const {
+    fullbleedContainer,
+    eyedropperActive,
+    eyedropperCallback,
+    eyedropperImg,
+    eyedropperPixelData,
+  } = useCanvas(
+    ({
+      state: {
+        fullbleedContainer,
+        eyedropperActive,
+        eyedropperCallback,
+        eyedropperImg,
+        eyedropperPixelData,
+      },
+    }) => ({
+      fullbleedContainer,
+      eyedropperActive,
+      eyedropperCallback,
+      eyedropperImg,
+      eyedropperPixelData,
+    })
+  );
+
+  const { pageWidth } = useLayout(({ state: { pageWidth } }) => ({
+    pageWidth,
+  }));
+
+  const fullHeight = pageWidth / FULLBLEED_RATIO;
+  const img = eyedropperImg;
+  const imgRef = useRef();
+  const zoomCanvas = useRef();
```

Comment:
I wonder if there's a better name for it instead of `zoom` so that it wouldn't get confusing with the actual zoom feature in the editor? E.g. perhaps `magnifier` or sth similar. WDYT?